engines/libaio.c

```diff
@@ -366,27 +444,26 @@ static int fio_libaio_init(struct thread_data *td)
 	ld->io_us = calloc(ld->entries, sizeof(struct io_u *));
 
 	td->io_ops_data = ld;
+	/*
+	 * Initialize prio_percent mechanisms if relevant
+	 */
+	if (!(fio_option_is_set(to, ioprio) ||
+			fio_option_is_set(to, ioprio_class) ||
+			(o->prio_percent == 0))) {
+		ioengine.prio_prep = fio_libaio_prio_prep;
+		ld->prio_seed = FIO_RANDSEED * td->thread_number
+				+ FIO_RAND_NR_OFFS;
+		init_rand_seed(&ld->prio_state, ld->prio_seed, false);
+	}
+	else if (o->prio_percent != 0) {
+		log_err("%s: prio_percent option and mutually exclusive "
+				"prio or prioclass option is set, exiting\n", to->name);
+		td_verror(td, EINVAL, "fio_libaio_init");
+		return 1;
```

Comment:
Should this be mentioned in the help?

Reply:
Yes, I'll add that to the prio_percent description.